src/main/java/com/github/manolo8/darkbot/utils/login/LoginUtils.java

```diff
@@ -61,41 +62,57 @@ public static LoginData performUserLogin(StartupParams params) {
         return loginData;
     }
 
-    public static LoginData performAutoLogin(StartupParams params) {
+    public static LoginData performAutoLogin(StartupParams.AutoLoginProps params) {
         String username = params.getAutoLoginValue(StartupParams.PropertyKey.USERNAME);
         String password = params.getAutoLoginValue(StartupParams.PropertyKey.PASSWORD);
-
-        if (username != null && (password == null || password.isEmpty())) {
+        String server = params.getAutoLoginValue(StartupParams.PropertyKey.SERVER);
+        String sid = params.getAutoLoginValue(StartupParams.PropertyKey.SID);
+        boolean isSidLogin = !Strings.isEmpty(sid) && !Strings.isEmpty(server);
+        if (!isSidLogin && username != null && Strings.isEmpty(password)) {
             password = getPassword(username, params.getAutoLoginMasterPassword());
 
             if (password == null)
                 System.err.println("Password for user couldn't be retrieved. Check that the user exists and master password is correct.");
         }
 
-        if (username == null || password == null || password.isEmpty()) {
-            System.err.println("Credentials file requires username and either a password or a master password");
+        if (!isSidLogin && (username == null || Strings.isEmpty(password))) {
+            System.err.println("Credentials file requires username & either a password or a master password, or/and server & sid");
             System.exit(-1);
         }
 
         LoginData loginData = new LoginData();
         loginData.setCredentials(username, password);
 
+        System.out.println("Attempt of auto logging using " + (isSidLogin?"server/SID":"login/password") + " in (1/2)");
+        if (isSidLogin) loginData.setSid(sid, server + ".darkorbit.com");
+        else usernameLogin(loginData);
```

Comment:
```suggestion
```
this needs to occur inside the try

the setSid is what can be here without an issue because it makes no calls that could lead to an exception, the usernameLogin **can** lead to exceptions and should be inside the try